tool/src/org/antlr/v4/codegen/CodeGenPipeline.java

```diff
@@ -50,6 +50,8 @@ public void process() {
 				writeRecognizer(parser, gen, false);
 			}
 
+			g.removeParserHeader();
```

Comment:
What's the meaning of such removing? It doesn't look clear.

Reply:
See #3868